### ISSUE TITLE
Or and And predicates only accept a single property

### ISSUE THREAD

**Issue by p013570** [gchq/Gaffer#877]
Since updating to use the Koryphe functions library in version 0.7.0 the Or and And Predicates now only work with a single property.

 For example you can select a property (e.g count) and pass that same property value to multiple predicates:

```java
new View.Builder()
        .edge("RedEdge", new ViewElementDefinition.Builder()
                .preAggregationFilter(new ElementFilter.Builder()
                                .select("count")
                                .execute(new Or<>(new IsLessThan(2L), new IsMoreThan(10L)))
                                .build())
                .build())
        .build()
```

Prior to 0.7.0 you could select multiple properties then within the Or function you could do select which properties to pass to which functions, it was complicated to write but it used to work like this:
```java
new View.Builder()
        .edge("RedEdge", new ViewElementDefinition.Builder()
                .preAggregationFilter(new ElementFilter.Builder()
                                .select("count", "average")
                                .execute( new Or(new ConsumerFunctionContext.Builder<Integer, FilterFunction>()
                                                .select(0) // select first property
                                                .execute(new IsLessThan(2))
                                                .build(),
                                        new ConsumerFunctionContext.Builder<Integer, FilterFunction>()
                                                .select(1) // select second property
                                                .execute(new IsMoreThan(5.1))
                                                .build())))
                                .build())
                .build())
        .build()
```

**Comment by p013570** [comment]
The issue was actually mainly around the validation of the Or and And predicates. This has now been fixed in Koryphe 0.1.1. So to fix this issue we will need to update to Koryphe 0.1.1.

**Comment by p013570** [comment]
Merged into develop.
